core/tx_pool.go

```diff
@@ -1334,9 +1333,9 @@ func (pool *TxPool) runReorg(done chan struct{}, reset *txpoolResetRequest, dirt
 	if reset != nil {
 		pool.demoteUnexecutables()
 		if reset.newHead != nil && pool.chainconfig.IsCurie(new(big.Int).Add(reset.newHead.Number, big.NewInt(1))) {
-			l1BaseFee := fees.GetL1BaseFee(pool.currentState)
-			pendingBaseFee := misc.CalcBaseFee(pool.chainconfig, reset.newHead, l1BaseFee)
-			pool.priced.SetBaseFee(pendingBaseFee)
+			// note: we cannot predict the next block's base fee since it is set
+			// by the sequencer, so we use the current block's base fee instead.
+			pool.priced.SetBaseFee(reset.newHead.BaseFee)
```

Comment:
this makes sense for follower nodes but does it really matter? I assume this is only relevant for txpool, not estimations users will get.

This will make sequencer's txpool less reliable as well. It might start to give worker txns that are below baseFee for pending block.

Reply:
Makes sense. I reverted the change. If follower nodes' txpool is not accurate it's not so bad, what's important is that the signer node's is accurate.